src/coreclr/src/vm/appdomain.cpp

```diff
@@ -2002,6 +2003,11 @@ void SystemDomain::LoadBaseSystemClasses()
     MscorlibBinder::LoadPrimitiveType(ELEMENT_TYPE_I);
     MscorlibBinder::LoadPrimitiveType(ELEMENT_TYPE_U);
 
+    // further loading of nonprimitive types may need casting support.
+    // initialize cast cache here.
+    MscorlibBinder::GetClass(CLASS__CASTCACHE);
```

Comment:
Is this necessary? ` MscorlibBinder::GetField(FIELD__CASTCACHE__TABLE)` in CastCache::Initialize should take care of this.

Reply:
No, it will be loaded regardless. I just wanted to make it more explicit that we will be loading a type here. I can make it a comment.

Reply:
Number of other calls here (e.g. `CrossLoaderAllocatorHashSetup::EnsureTypesLoaded` a few lines below) load types without going into details.

Reply:
I'll just remove this line then.